core/src/main/resources/error/error-classes.json

```diff
@@ -170,7 +170,7 @@
       },
       "NON_FOLDABLE_INPUT" : {
         "message" : [
-          "the input should be a foldable string expression and not null; however, got <inputExpr>."
+          "the input <inputName> should be a foldable <inputType> expression; however, got <inputExpr>."
```

Comment:
if input is null, should change to use `UNEXPECTED_NULL`, else the inputType will be `Void`